### PR TITLE
Extend editor controller, support no category toolbox

### PR DESCRIPTION
**Pull request by samelhusseini**
Extend editor controller with new capabilities: 
- set Scale (zoom)
- workspace has loaded (signal)

Support no category toolbox (blockly loader). 

Minor cleanup work.

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -137,6 +137,9 @@ export class ProjectView
         // (e.g. from the URL hash or from WinRT activation arguments)
         const skipStartScreen = pxt.appTarget.appTheme.allowParentController
             || window.location.hash == "#editor";
+        if (pxt.appTarget.appTheme.allowParentController && pxt.shell.layoutTypeClass() != 'widget') {
```
[the remark this answers is not in the document]
Comment (samelhusseini) [reply webapp/src/app.tsx]: Not really. I'm pretty sure it's safe, if it's allowParentController then the index.html does nothing, you have to view it from an iframe. 
I guess it's possible to use something other than controller.html..

[the remark this answers is not in the document]
Comment (samelhusseini) [reply webapp/src/app.tsx]: Probably should remove it.

File: pxtwinrt/winrtworkspace.ts [hunk 2]
```diff
@@ -322,6 +322,10 @@ namespace pxt.winrt.workspace {
             }));
     }
 
+    function loadedAsync(): Promise<void> {
```
Comment (riknoll) [inline pxtwinrt/winrtworkspace.ts]: What is this about?

Comment (riknoll) [reply pxtwinrt/winrtworkspace.ts]: Ah, never mind. I get it

**Comment by pelikhan** [comment]
This is going stale.

**Comment by samelhusseini** [comment]
Ill take care of it today

**Pull request merged**